File: haskell/gigasecond/Gigasecond.hs
{-|
Module      : Gigasecond
Copyright   : (c) Eric Bailey, 2015
License     : MIT

Maintainer  : Eric Bailey
Stability   : experimental
Portability : portable

Calculating when someone turned or will turn 1 Gs old.
-}

module Gigasecond (
  fromDay
  -- * Usage
  -- $usage
  ) where

import Data.Time.Clock (UTCTime, addUTCTime)

-- | Given a time, returns the time a gigasecond later.
fromDay :: UTCTime -> UTCTime
fromDay = addUTCTime 1000000000

{- $usage
>>> import Data.Time.Format (readTime)
>>> import System.Locale (defaultTimeLocale, iso8601DateFormat)
>>> let fromString = readTime defaultTimeLocale (iso8601DateFormat (Just "%T%Z")) :: String -> UTCTime
2020-10-18 04:21:40 UTC
-}

File: go/gigasecond/gigasecond.go
// Package gigasecond implements the [Gigasecond] exercise.
//
// [Gigasecond]: https://exercism.org/tracks/go/exercises/gigasecond
package gigasecond

import (
	"time"
)

// Gigasecond is a billion seconds in nanoseconds.
const Gigasecond time.Duration = 1e9 * 1e9

// AddGigasecond returns the time.Time a gigasecond later than a given
// time.Time.
func AddGigasecond(t time.Time) time.Time {
	return t.Add(Gigasecond)
}
